### ISSUE TITLE
ImathTest fails for x86 32-bit platform

### ISSUE THREAD

**Issue by waebbl** [AcademySoftwareFoundation/openexr#346]
Hi, I'm co-maintainer for the openexr related packages on Gentoo Linux.

I'm using a multilib platform, allowing 64-bit and 32-bit applications to run in parallel on a x86_64 and using gcc version 7.3.0. IlmBase compiles just fine for both arches, but when I run the testsuite it finishes successfully for 64-bit, but fails for 32-bit. I also tested it in a 32-bit only chroot enviroment with the same result.

It errors on ImathTest with a core dump with
```
/tmp/portage/media-libs/ilmbase-2.3.0/work/ilmbase-2.3.0/test-driver: line 95: 26717 Aborted                 (core dumped) "$@" > $log_file 2>&1
FAIL: ImathTest
```

test-suite.log contains the info
```
ImathTest: /tmp/portage/media-libs/ilmbase-2.3.0/work/ilmbase-2.3.0/ImathTest/testBoxAlgo.cpp:892: void {anonymous}::boxMatrixTransform(): Assertion `b21 == b2' failed.
```
Any idea, what's the cause for this failure?
TIA

**Comment by malaterre** [comment]
The title of the issue is a bit misleading. It should be refer to x86 arch instead (x87 double rounding issue). This issue is not visible for instance on ppc, mips or arm (although 32bits architectures).

**Comment by malaterre** [comment]
Here is a silly patch:

* https://bugs.debian.org/909865#14

Comments welcome

**Comment by waebbl** [comment]
Sorry, I totally lost track of this issue.
We're in the process on moving to 2.4 release on Gentoo, where I didn't encounter this problem anymore. On my tests so far, all test cases succeeded.
Therefore, from my point of view, this issue can be closed.

**Comment by cary-ilm** [comment]
Closing the issue for now, feel free to re-open or file a new issue if you need further help.

**Comment by waebbl** [comment]
> Here is a silly patch:
> 
> * https://bugs.debian.org/909865#14
> 
> Comments welcome

This issue popped up again on x86 platform. However, by adding `-ffloat-store` from @malaterre's patch to CPPFLAGS for this platform solves it.
Thanks for the patch.
